ext/sftp/sink.go

```diff
@@ -168,5 +196,99 @@ func (s *SFTPSink) process() error {
 		}
 		return nil
 	})
-	return errors.WithStack(err)
+	if err != nil {
+		s.Logger().Error(fmt.Sprintf("failed to flush data: %s", err.Error()))
+		return errors.WithStack(err)
+	}
+
+	if s.enableArchive {
+		err := s.DryRunable(func() error {
+			pathsToArchive := []string{}
+			for destinationURI := range s.writerHandlers {
+				tmpPath, err := getTmpPath(destinationURI)
+				if err != nil {
+					s.Logger().Error(fmt.Sprintf("failed to get tmp path for %s: %s", destinationURI, err.Error()))
+					return errors.WithStack(err)
+				}
+
+				pathsToArchive = append(pathsToArchive, tmpPath)
+			}
+			s.Logger().Info(fmt.Sprintf("compressing %d files: %s", len(pathsToArchive), strings.Join(pathsToArchive, ", ")))
+
+			archivePaths, err := s.archive(pathsToArchive)
+			if err != nil {
+				s.Logger().Error(fmt.Sprintf("failed to compress files: %s", err.Error()))
+				return errors.WithStack(err)
+			}
+
+			s.Logger().Info(fmt.Sprintf("successfully uploaded archive file(s) to SFTP: %s", strings.Join(archivePaths, ", ")))
+
+			return nil
+		})
+		if err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	return nil
+}
+
+func getTmpPath(destinationURI string) (string, error) {
+	targetURI, err := url.Parse(destinationURI)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return filepath.Join("/tmp", filepath.Base(targetURI.Path)), nil
+}
+
+func (s *SFTPSink) archive(filesToArchive []string) ([]string, error) {
```

Comment:
as the archive function is repetitive across sinks, we can do refactor later

Reply:
indeed